components/brave_wallet/browser/eth_data_builder_unittest.cc

```diff
@@ -259,6 +270,70 @@ TEST(EthCallDataBuilderTest, Addr) {
             "43fcd34d8589090581e1d2bdcf5dc17feb05b2006401fb1c3fdded335a465b51");
 }
 
+TEST(EthCallDataBuilderTest, DnsEncode) {
+  // Based on DNSUtilTest.DNSDomainFromDot test. But without total length limit
+  // and support of terminal dot.
+
+  EXPECT_FALSE(DnsEncode(""));
+  EXPECT_FALSE(DnsEncode("."));
+  EXPECT_FALSE(DnsEncode(".."));
+  EXPECT_FALSE(DnsEncode("foo,bar.com"));
+
+  EXPECT_EQ(DnsEncode("com"), VectorFromBinaryString("\003com"));
+  EXPECT_EQ(DnsEncode("google.com"),
+            VectorFromBinaryString("\x006google\003com"));
+  EXPECT_EQ(DnsEncode("www.google.com"),
+            VectorFromBinaryString("\003www\006google\003com"));
+
+  // Label is 63 chars: still valid
+  EXPECT_EQ(
+      DnsEncode(
+          "z23456789a123456789a123456789a123456789a123456789a123456789a123"),
+      VectorFromBinaryString(
+          "\077z23456789a123456789a123456789a123456789a123456"
+          "789a123456789a123"));
+
+  // Label is too long: invalid
+  EXPECT_FALSE(DnsEncode(
+      "123456789a123456789a123456789a123456789a123456789a123456789a1234"));
+
+  // 253 characters in the name: still valid
+  EXPECT_EQ(DnsEncode("abcdefghi.abcdefghi.abcdefghi.abcdefghi.abcdefghi."
+                      "abcdefghi.abcdefghi."
+                      "abcdefghi.abcdefghi.abcdefghi.abcdefghi.abcdefghi."
+                      "abcdefghi.abcdefghi."
+                      "abcdefghi.abcdefghi.abcdefghi.abcdefghi.abcdefghi."
+                      "abcdefghi.abcdefghi."
+                      "abcdefghi.abcdefghi.abcdefghi.abcdefghi.abc"),
+            VectorFromBinaryString(
+                "\011abcdefghi\011abcdefghi\011abcdefghi\011abcdefghi\011a"
+                "bcdefghi\011abcdefghi\011abcdefghi\011abcdefghi\011abcdef"
+                "ghi\011abcdefghi\011abcdefghi\011abcdefghi\011abcdefghi"
+                "\011abcdefghi\011abcdefghi\011abcdefghi\011abcdefghi\011a"
+                "bcdefghi\011abcdefghi\011abcdefghi\011abcdefghi\011abcdef"
+                "ghi\011abcdefghi\011abcdefghi\011abcdefghi\003abc"));
+
+  // 254 characters in the name: still valid.
+  // Per ENS spec there is no total lenght limitation.
+  // https://docs.ens.domains/ens-improvement-proposals/ensip-10-wildcard-resolution#specification
+  EXPECT_TRUE(DnsEncode(
+      "123456789.123456789.123456789.123456789.123456789.123456789.123456789."
+      "123456789.123456789.123456789.123456789.123456789.123456789.123456789."
+      "123456789.123456789.123456789.123456789.123456789.123456789.123456789."
+      "123456789.123456789.123456789.123456789.1234"));
```

Comment:
Are `IP` alike valid DnsEncoded ENS?

Reply:
Technically DnsEncode of an ipaddress will result in something meaningful. I don't any restriction why it shouldn't.
In our case all ENS stuff happens for .eth tld.
In [reference implementation](https://github.com/ensdomains/offchain-resolver/blob/main/packages/gateway/src/server.ts#L30-L40 ) gateway is interpreting that encoded name to extract required record. Makes most sense for wildcard case which is not part of this PR

Actually I'm not sure I understand why ENS designers brought DNSEncoding here - plain string would be enough. Now it looks to me as unjustified complication.